ifrs17/Import/Importers.ipynb

```diff
@@ -310,50 +310,106 @@
+                "\n}"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public async Task<string> GetReportingNodeFromMainAsync(IDataSet dataSet)",
+                "\n{",
+                "\n    var mainTab = dataSet.Tables[Main];",
+                "\n    if(mainTab.Columns.FirstOrDefault(x => x.ColumnName == nameof(ReportingNode)) == null) ApplicationMessage.Log(Error.ReportingNodeInMainNotFound);",
                 "\n    if(ApplicationMessage.HasErrors()) return null;",
-                "\n",
-                "\n    ImportArgs args;",
+                "\n    return (string)mainTab.Rows.First()[nameof(ReportingNode)];",
+                "\n}"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public async Task<PartitionedArgs> GetPartitionedArgsFromMainAsync(IDataSet dataSet)",
+                "\n{",
+                "\n    var args = await GetArgsFromMainAsync(dataSet);",
+                "\n    var mainTab = dataSet.Tables[Main];",
+                "\n    if(mainTab.Columns.FirstOrDefault(x => x.ColumnName == nameof(ReportingNode)) == null) ApplicationMessage.Log(Error.ReportingNodeInMainNotFound);",
+                "\n    if(ApplicationMessage.HasErrors()) return null;",
+                "\n    var reportingNode = await GetReportingNodeFromMainAsync(dataSet);",
+                "\n    return new PartitionedArgs(reportingNode, args.Year, args.Month, default(Periodicity), args.Scenario);",
+                "\n}"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "markdown",
+            "source": [
+                "## Parse Main and return ImportArgs",
+                "\n   ",
+                "\nThe main table of our custom import formats contains the information which are required to identify the data partition. These information are temporarily stored in [PartitionedArgs](../DataModel/DataStructure#partitionedargs) and used in the next methods."
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public async Task<ImportArgs> GetArgsFromMainAsync<IPartition>(IDataSet dataSet)",
+                "\n{",
+                "\n    ImportArgs importArgs;",
                 "\n    switch(typeof(IPartition).Name) {",
                 "\n        case nameof(PartitionByReportingNode) : {",
-                "\n            args = new ImportArgs( reportingNode,",
+                "\n            var reportingNode = await GetReportingNodeFromMainAsync(dataSet);",
```

Comment:
this method misses this validation: 
"var mainTab = dataSet.Tables[Main];",
"\n    if(mainTab == null) ApplicationMessage.Log(Error.NoMainTab);",

Imo it could live in GetArgsFromMainAsync method.